Base/QTGUI/qSlicerSaveDataDialog.cxx

```diff
@@ -806,7 +814,25 @@ bool qSlicerSaveDataDialogPrivate::saveNodes()
 
     // file properties
     QFileInfo file = this->file(row);
-    QString format = this->format(row);
+    QString format = this->format(row).toLower(); // Convert format to lowercase for comparison
+
+    // Check if the format is TIFF, show warning if necessary, and only once
+    if (!tiffWarningShown && (format.contains(".tiff") || format.contains(".tif")))
+    {
+      QMessageBox::StandardButton reply = QMessageBox::warning(this, "TIFF Format Warning",
+        "Saving 3D images in TIFF format may result in loss of orientation and Z-spacing data. "
+        "Consider using .nrrd format for optimal preservation. TIFF is not suitable for 3D images.\n"
+        "Do you want to continue?",
```

Comment:
It would be great if you could mark these strings for translation.  You could follow the pattern i the rest of the code and there are more details here: https://github.com/Slicer/SlicerLanguagePacks/blob/main/DevelopersManual.md

Reply:
I changed some code. Now, I'm not sure if this request still is relevant. I checked similar parts of that method, and they don't use `tr()`. Should I still add it here? Let me know if I should

Reply:
It's not strictly required but we'll want to do it someday.  When people are working on formatting strings it can be easier to add the translation hooks while everything is fresh.  Otherwise someone later needs to make sure they understand the logic.